### PR TITLE
FEATURE: EGL-22 - add the configuration to display the continuous line for the null values

### PR DESCRIPTION
**Pull request by vannguyengd**
feat: connect missing values
JIRA: EGL-22

---

Supported PR commands:

| Command                                                 | Description                                                |
| ------------------------------------------------------- | ---------------------------------------------------------- |
| `ok to test`                                            | Re-run standard checks                                     |
| `extended check sonar`                                  | SonarQube tests                                            |
| `extended test - backstop`                              | BackstopJS tests                                           |
| **E2E Cypress tests commands - TIGER**                  |                                                            |
| `extended test - tiger-cypress - isolated <testName>`   | Run isolated tests running against recorded Tiger backend. |
| `extended test - tiger-cypress - record <testName>`     | Create a new recording for isolated Tiger tests.           |
| `extended test - tiger-cypress - integrated <testName>` | Run integrated tests against live backend                  |
| **E2E Cypress tests commands - BEAR**                   |                                                            |
| `extended test - cypress - isolated <testName>`         | Run isolated tests running against recorded Bear backend.  |
| `extended test - cypress - record <testName>`           | Create a new recording for isolated Bear tests.            |
| `extended test - cypress - integrated <testName>`       | Run integrated tests against live backend                  |

`<testName>` in cypress commands is used to filter specfiles. Example, to run record with BEAR backend

-   Against `dashboard.spec.ts` and `drilling.spec.ts`, execute command `extended test - cypress - record dashboard,drilling`
-   Against all specfiles, execute command `extended test - cypress - record` or `extended test - cypress - record *`

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `extended test - backstop` passes
-   [ ] `extended test - tiger-cypress - record` to record new mapping files (Tiger BE)
-   [ ] `extended test - cypress - record` to record new mapping files (Bear BE)
-   [ ] `extended test - tiger-cypress - isolated` passes
-   [ ] `extended test - cypress - isolated` passes
-   [ ] `extended test - tiger-cypress - integrated` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/12896/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/14709/ : SUCCESS in 11m 37s

**Comment by vannguyengd** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/12899/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/14712/ : SUCCESS in 11m 28s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/12900/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/14713/ : SUCCESS in 11m 29s

File: libs/sdk-ui-ext/src/internal/components/configurationPanels/LineChartBasedConfigurationPanel.tsx [hunk 1]
```diff
@@ -86,4 +103,20 @@ export default class LineChartBasedConfigurationPanel extends BaseChartConfigura
...
     }
+
+    private isContinuousLineSelectable(): boolean {
+        const { insight, properties, type } = this.props;
+        let isStackingMeasures = properties?.controls?.stackMeasures;
+        if (isStackingMeasures) {
+            return false;
+        } else if (typeof isStackingMeasures === "undefined" && isAreaChart(type)) {
+            isStackingMeasures =
+                insight.insight.buckets.find((item) => item.localIdentifier === BucketNames.MEASURES)?.items
```
Comment (pbenes) [inline libs/sdk-ui-ext/src/internal/components/configurationPanels/LineChartBasedConfigurationPanel.tsx:114]: Please check, maybe this https://github.com/gooddata/gooddata-ui-sdk/blob/master/libs/sdk-ui-ext/src/internal/utils/bucketRules.ts#L67 or some other similar helper here could be handy.

File: libs/sdk-ui-charts/api/sdk-ui-charts.api.md [hunk 2]
```diff
@@ -189,6 +189,7 @@ export interface IChartConfig {
     colorMapping?: IColorMapping[];
     colorPalette?: IColorPalette;
     colors?: string[];
+    continuousLine?: IContinuousLineConfig;
```
Comment (pbenes) [inline libs/sdk-ui-charts/api/sdk-ui-charts.api.md:192]: Please prepare storybook screenshot test for this configuration. For line chart with this option enabled, and consider also having one screenshot for combo chart with this config (optional).

File: libs/sdk-ui-ext/src/internal/components/pluggableVisualizations/comboChart/PluggableComboChart.tsx [hunk 3]
```diff
@@ -370,6 +381,7 @@ export class PluggableComboChart extends PluggableBaseChart {
                     axis={this.axis}
                     panelConfig={{
                         isDataPointsControlDisabled: this.isDataPointsControlDisabled(insight),
+                        isContinuousLineControlDisabled: this.isContinuousLineControlDisabled(insight),
```
[the remark this answers is not in the document]
Comment (vannguyengd) [reply libs/sdk-ui-ext/src/internal/components/pluggableVisualizations/comboChart/PluggableComboChart.tsx:384]: Yes, we will disable the continuous line checkbox when the combo chart only has column series.

File: libs/sdk-ui-ext/src/internal/components/configurationPanels/LineChartBasedConfigurationPanel.tsx (hunk 1 above)
Comment (pbenes) [reply libs/sdk-ui-ext/src/internal/components/configurationPanels/LineChartBasedConfigurationPanel.tsx:114]: Ah, that's a pity, what about:
```
import { insightBucket } from "@gooddata/sdk-model";
import { BucketNames } from "@gooddata/sdk-ui";
...
        const measuresBucket = insightBucket(insight, BucketNames.MEASURES);
        const hasMoreThanOneMetric = measuresBucket.items.length > 1;
```
Please check if that could work, thanks.

File: libs/sdk-ui-ext/src/internal/translations/en-US.json [hunk 4]
```diff
@@ -706,6 +706,16 @@
         "comment": "applicable for funnel chart special data labels which can include percentages",
         "limit": 0
     },
+    "properties.canvas.continuousLine.label": {
+        "value": "Continuous line",
+        "comment": "",
+        "limit": 0
+    },
+    "properties.canvas.continuousLine.tooltip": {
+        "value": "Draw a dashed line between the points with missing values.",
```
Comment (phong-nguyen-duy) [inline libs/sdk-ui-ext/src/internal/translations/en-US.json:715]: The tooltip should be "Draw a solid line between points with missing values"

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/12960/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/14766/ : FAILURE in 15m 07s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/12962/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/14768/ : FAILURE in 15m 05s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/12963/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/14769/ : FAILURE in 15m 56s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/12964/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/14770/ : FAILURE in 15m 34s

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/13013/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/14811/ : FAILURE in 15m 44s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: libs/sdk-ui-ext/src/internal/translations/en-US.json (hunk 4 above)
Comment (vannguyengd) [reply libs/sdk-ui-ext/src/internal/translations/en-US.json]: I have fixed this.  Thanks!

File: libs/sdk-ui-ext/src/internal/components/configurationPanels/LineChartBasedConfigurationPanel.tsx (hunk 1 above)
Comment (vannguyengd) [reply libs/sdk-ui-ext/src/internal/components/configurationPanels/LineChartBasedConfigurationPanel.tsx]: I have updated this. Thanks!

File: libs/sdk-ui-charts/api/sdk-ui-charts.api.md (hunk 2 above)
Comment (vannguyengd) [reply libs/sdk-ui-charts/api/sdk-ui-charts.api.md:192]: I have added the storybook test. Thanks!!

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/13015/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/14813/ : SUCCESS in 15m 41s

**Comment by vannguyengd** [comment]
extended test - backstop

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/2079/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/13016/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-pull-request-dispatcher-pipeline/523/ : UNSTABLE in 23m 57s

**Comment by vannguyengd** [comment]
extended test - backstop

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/13018/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/14815/ : SUCCESS in 15m 40s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/2080/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/13019/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-pull-request-dispatcher-pipeline/524/ : SUCCESS in 23m 35s

**Review by pbenes — APPROVED** [review]

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR3687

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/2654/ : SUCCESS in 11m 07s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/535/ : FAILURE in 1m 15s (non-voting)
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/14826/ : SUCCESS in 15m 41s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tiger-isolated-zuul-pipeline/139/ : SUCCESS in 8m 26s (non-voting)
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul-pipeline/177/ : SUCCESS in 10m 36s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/3809/ : SUCCESS in 13s

**Pull request merged**